spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/test/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfigurationTests.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.sql.autoconfig;
+
+import java.io.IOException;
+
+import javax.sql.DataSource;
+
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.api.services.sqladmin.model.DatabaseInstance;
+import com.zaxxer.hikari.HikariDataSource;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.gcp.core.autoconfig.GcpContextAutoConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author João André Martins
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+		classes = {GcpCloudSqlAutoConfiguration.class, GcpContextAutoConfiguration.class},
+		properties = {"spring.cloud.gcp.projectId=proj",
+				"spring.cloud.gcp.sql.instanceName=test-instance",
+				"spring.cloud.gcp.sql.databaseName=test-database",
+				"spring.cloud.gcp.sql.userName=watchmaker",
+				"spring.cloud.gcp.sql.password=pass"}
+)
+public class GcpCloudSqlAutoConfigurationTests {
+
+	@Autowired
+	private DataSource dataSource;
+
+	@MockBean
+	private SQLAdmin mockSqlAdmin;
+
+	@Mock
+	private SQLAdmin.Instances mockInstances;
+
+	@Mock
+	private SQLAdmin.Instances.Get mockGet;
+
+	private DatabaseInstance databaseInstance = new DatabaseInstance();
+
+	@Before
+	public void setUp() throws IOException {
```

Comment:
It only works because I applied that "hack" I mentioned earlier -- I'm mocking the `SQLAdmin` behaviour in the configuration, not in the `@Before` `setUp()` method as is best done.